### PR TITLE
feat(nodejs): add dependency line numbers for npm lock files

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
Add dependency line numbers for npm lock files

e.g.:
package-lick.json file:
```zsh
{
  "name": "intern-angular",
  "version": "1.0.0",
  "lockfileVersion": 1,
  "dependencies": {
    "@angular/animations": {
      "version": "4.2.6",
      "resolved": "[https://registry.npmjs.org/@angular/animations/-/animations-4.2.6.tgz"](https://registry.npmjs.org/@angular/animations/-/animations-4.2.6.tgz%22),
      "integrity": "sha1-nZyAoRmwwDaTy9I7uvcosVMf/8c="
    }
}
```

Trivy json result(with `--list-all-pkgs` flag):
```zsh
➜ ./trivy fs -f json --list-all-pkgs ./package-lock.json
...
  "Results": [
    {
      "Target": "package-lock.json",
      "Class": "lang-pkgs",
      "Type": "npm",
      "Packages": [
        {
          "ID": "@angular/animations@4.2.6",
          "Name": "@angular/animations",
          "Version": "4.2.6",
          "Indirect": true,
          "Layer": {},
          "Locations": [
            {
              "StartLine": 6,
              "EndLine": 10
            }
          ]
        }
      ]
...
```

## Related PRs
- [x] https://github.com/aquasecurity/go-dep-parser/pull/134

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/fanal/analyzer/language/analyze.go [hunk 1]
```diff
@@ -42,6 +42,16 @@ func ToAnalysisResult(fileType, filePath, libFilePath string, libs []godeptypes.
...
+		var locs []types.Location
+		if len(lib.Locations) != 0 {
+			for _, loc := range lib.Locations {
+				l := types.Location{
+					StartLine: loc.StartLine,
+					EndLine:   loc.EndLine,
+				}
+				locs = append(locs, l)
+			}
+		}
```
Comment (knqyf263) [inline pkg/fanal/analyzer/language/analyze.go:54]: Looks like we don't need the if-statement.

```suggestion
		for _, loc := range lib.Locations {
			l := types.Location{
				StartLine: loc.StartLine,
				EndLine:   loc.EndLine,
			}
			locs = append(locs, l)
		}
```

File: pkg/fanal/analyzer/language/nodejs/npm/testdata/wrong.json [hunk 2]
```diff
@@ -0,0 +1,68 @@
+{
```
Comment (knqyf263) [inline pkg/fanal/analyzer/language/nodejs/npm/testdata/wrong.json:1]: If we just want to test a case where JSON cannot be parsed, can we make this JSON small? For example, `{broken}` is an invalid JSON.

File: pkg/fanal/analyzer/language/analyze.go (hunk 1 above)
Comment (DmitriyLewen) [reply pkg/fanal/analyzer/language/analyze.go]: Thanks, Fixed!

File: pkg/fanal/analyzer/language/nodejs/npm/testdata/wrong.json (hunk 2 above)
Comment (DmitriyLewen) [reply pkg/fanal/analyzer/language/nodejs/npm/testdata/wrong.json]: Done

**Review by knqyf263 — APPROVED** [review]

**Pull request merged**